keps/sig-node/4188-kubelet-pod-readiness-api/README.md

```diff
@@ -217,75 +240,90 @@ the Pods. We don't want this API to return partial data.
 
 ## Design Details
 
+### Pod State Selection
+
+Kubelet maintains multiple representations of pod state:
+- The state derived from the Container Runtime Interface (CRI), reflecting the actual status of containers on the node.
+- The state that is sent to the API server, which may be subject to additional processing or delays.
+- The state received from the API server, reflecting the control plane's view.
+
+It is important to define which state this API will serve. For the purposes of this API, the intent is to serve the most
+up-to-date and accurate pod state as known locally by the Kubelet, typically the state based on the CRI and Kubelet's internal
+reconciliation, rather than the potentially stale state received from the API server. This ensures consumers receive the
+freshest possible information about pods running on the node.
+
 ### Proposed API
 
-We propose to add new gPRC API `status` in Kubelet, listening on a unix socket
-at `/var/lib/Kubelet/status/Kubelet.sock`. The endpoint will be versioned. The
-gRPC Service will expose 3 methods serving local Pods statuses data:
+We propose to add new gRPC API `pods` in Kubelet, listening on a unix socket
+at `/var/lib/Kubelet/pods/Kubelet.sock`. The endpoint will not be versioned. The
+gRPC Service will expose 3 methods serving local Pods data:
 
 ```protobuf
-service PodStatus {
-    // ListPodStatus returns a of List of PodStatus
-    rpc ListPodStatus(PodStatusListRequest) returns (PodStatusListResponse) {}
-    // WatchPodStatus returns a stream of List of PodStatus
-    // Whenever a pod state change api returns the new list
-    rpc WatchPodStatus(PodStatusWatchRequest) returns (stream PodStatusWatchResponse) {}
-    // GetPodStatus returns a PodStatus for given pod's UID
-    rpc GetPodStatus(PodStatusGetRequest) returns (PodStatusGetResponse) {}
+import "google/protobuf/field_mask.proto";
+
+service Pods {
+    // ListPods returns a list of v1.Pod, filtered by field mask.
+    rpc ListPodStatus(PodListRequest) returns (PodListResponse) {}
+    // WatchPods returns a stream of list of PodInfo, filtered by field mask.
+    // Whenever a pod state changes, api returns the new list.
+    rpc WatchPods(PodWatchRequest) returns (stream PodWatchResponse) {}
+    // GetPod returns a PodInfo for given pod's UID, filtered by field mask.
+    rpc GetPod(PodGetRequest) returns (PodGetResponse) {}
 }
 
-// PodCondition aligns with v1.PodCondition.
-message PodCondition {
-    PodConditionType Type = 1;
-    ConditionStatus Status = 2;
-    Timestamp LastProbeTime = 3;
-    Timestamp LastTransitionTime = 4;
-    string Reason = 5;
-    string Message = 6;
+// PodListRequest allows specifying a field mask.
+message PodListRequest {
+    google.protobuf.FieldMask field_mask = 1;
```

Comment:
Google has actually [deprecated including fieldmasks in the request itself.](https://google.aip.dev/157).  Instead, clients are expected to include it in the gRPC metadata.

Reply:
ok, do you know how I should change this code snippet? or maybe I just remove it?